src/utils/auth.jsx

```diff
@@ -0,0 +1,45 @@
+import { baseUrl } from "../const";
+import { validateFetch } from "./api";
+
+const signup = ({ name, avatar, email, password }) => {
+  return fetch(`${baseUrl}/signup`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ name, avatar, email, password }),
+  }).then(validateFetch);

```

Comment:
**COULD BE IMPROVED**

You can make a special function for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(validateFetch)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.

Reply:
I can implement this in the future